src/app/models/store.model.ts

```diff
@@ -0,0 +1,10 @@
+import { StoreTagModel } from "./store-tag.model";
+
+export interface StoreModel {
+  readonly id: string;
+  readonly name: string;
+  readonly logoUrl: string;
+  readonly distanceInMeters: number;
+  readonly tagIds: string[];
+  readonly tags?: StoreTagModel[];
```

Comment:
i think we are mixing the query and data model here. Please make sure to separate it later. Ideally models would never have optional parameters (99% cases)